public/player.js

```diff
@@ -35,32 +35,42 @@
     YT.ready(onYouTubeIframeAPIReady);
   };
 
-  function toggleHighlight(x) {
-    x.classList.toggle("highlighted-option")
-    setTimeout(function() {
-      x.classList.remove("highlighted-option")
-    }, 3000)
+  var funcCount = (function () {
```

Comment:
@des-des Indeed, we are counting how many times the `onPlayerStateChange` function is invoked 

I want the user to be able to click on the right answer only after all the videos have finished playing. According to the YouTube IFrame Player API docs:

```The onStateChange event fires whenever the player's state changes. The data property of the event object that the API passes to your event listener function will specify an integer that corresponds to the new player state. Possible values are:
-1 (unstarted)
0 (ended)
1 (playing)
2 (paused)
3 (buffering)
5 (video cued)```

Checking if the last video from the playlist has finished playing or not, ended up being more complicated than I thought. By reading their docs you'd think that the new player state would be 0 only after the video has finished playing. However, the player state is also 0 before the video starts playing. So it is very odd that for the very same video, the player state is 0 twice. I thought it would be 0 once and only after the video finishes playing. This seems to happen only for the last video in the playlist. By console logging the current index of the video playing and `event.data` I realised that `onPlayerStateChange` is invoked 5 times only for the last video in the playlist. For the first video it gets invoked 3 times and for all the other videos 4 times. So the only way I could come up with to check if the last video from the playlist has finished playing, was to count how many times `onPlayerStateChange` has been invoked. Hence, the `funcCount` name for my function.